examples/tox21/roc_auc_evaluator.py

```diff
@@ -0,0 +1,52 @@
+import numpy
+
+import chainer
+from chainer import cuda
+from chainer.dataset import convert
+from chainer.training.extensions import Evaluator
+from sklearn import metrics
+
+
+def get_1d_numpy_array(v):
+    if isinstance(v, chainer.Variable):
+        v = v.data
+    return cuda.to_cpu(v).ravel()
+
+
+class ROCAUCEvaluator(Evaluator):
+
+    def __init__(self, iterator, target, predictor=None,
+                 converter=convert.concat_examples,
+                 device=None, eval_hook=None, eval_func=None, name=None):
+        super(ROCAUCEvaluator, self).__init__(
+            iterator, target, converter=converter, device=device,
+            eval_hook=eval_hook, eval_func=eval_func)
+        if name is not None:
+            self.name = name or 'validation'
+        self.predictor = predictor
+
+    def evaluate(self):
+        iterator = self._iterators['main']
+        predictor = self.predictor or self._targets['main']
+
+        iterator.reset()
+
+        y_total = []
+        t_total = []
+        for batch in iterator:
+            in_arrays = self.converter(batch, self.device)
+            with chainer.no_backprop_mode(), chainer.using_config('train',
+                                                                  False):
+                y = predictor(*in_arrays[:-1])
+            t = in_arrays[-1]
+            y_data = get_1d_numpy_array(y)
+            t_data = get_1d_numpy_array(t)
+            y_data = y_data[t_data != -1]
```

Comment:
I think this line is mistake.  In tox21 dataset we should ignore data point with -1 label because it means unlabeled (in chainer-chemistry). But ROC-AUC score function can be used for general purpose and scikit learn roc_auc_score can recieve -1 label data (https://github.com/scikit-learn/scikit-learn/blob/a24c8b46/sklearn/metrics/ranking.py#L465-L466). L45 is also.

Reply:
I read it, I think it means you can use either of the case

1. positive label is 1 and negative label is 0
2. positive label is 1 and negative label is -1
3. positive label is `pos_label` if specified, all other values are considered as negative labels.

So I think there is no way to "ignore" not to include from scoring.
Thus we need to include it.